transaction/AdmissionControl.hpp

```diff
@@ -0,0 +1,83 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ **/
+
+#ifndef QUICKSTEP_TRANSACTION_ADMISSION_CONTROL_HPP_
+#define QUICKSTEP_TRANSACTION_ADMISSION_CONTROL_HPP_
+
+#include "utility/Macros.hpp"
+#include "transaction/TransactionTable.hpp"
+
+namespace quickstep {
+namespace transaction {
+
+class AdmissionControl {
+ public:
+  /**
+   * @brief Constructor
+   * @param transaction_table A lookup table that stores information about
+   *        all the running and waiting transactions.
+   */
+  AdmissionControl(TransactionTable *transaction_table) {}
```

Comment:
Add `explicit`.

I was wondering which object will own `transaction_table`. And does this class need to store the pointer?

Also, should this class contain the `CompatibilityChecker` object?